docs/proposals/modifying-output-image.md

```diff
@@ -0,0 +1,126 @@
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-04-14
+last-updated: 2021-04-14
+status: provisional
+---
+
+# Modifying Output Image
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions
+
+1. Should this feature also allow users to _unset_ existing labels/annotations inherited from base images?
+
+## Summary
+
+Allow Build authors to describe modifications they'd like to make to output images.
+For this proposal, we'll scope this down to staticly-defined image [labels](https://docs.docker.com/engine/reference/builder/#label) and [OCI annotations](https://github.com/opencontainers/image-spec/blob/master/annotations.md).
+
+Future proposals might allow Build authors to configure image tags, and/or to allow authors to configure dynamic values based on input values.
+These are non-goals of this proposal.
+
+## Motivation
+
+### Goals
+
+- Begin to enable simple modifications of the output image by Build authors.
+
+### Non-Goals
+
+- Enable dynamic values or placeholders/templating (e.g., git commit SHA)
+- Automatically inject labels or annotations without user input
+- Enable modification of image layer contents (see [runtime image](https://github.com/shipwright-io/build/blob/master/docs/proposals/runtime-image.md))
+
+## Proposal
+
+Add new fields to describe output image labels and annotations:
+
+```yaml=
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: my-build-name
+spec:
+  output:
+    image: quay.io/my/image
+    credentials:
+      name: my-push-creds
+    # New fields
+    labels:
+      maintainer: team@my-company.com
+      description: "This is my cool image"
+    annotations:
+      org.opencontainers.image.url: https://my-company.com/images
+      org.opencontainers.image.source: https://github.com/org/repo
+```
+
+This will be enabled by embedding the existing `Image` struct in a new struct that 
+
+```go=
+type BuildSpec struct {
+    ...
+    Output OutputImage `json:"output"`
+}
+
+type OutputImage struct {
+    Image // embedding existing type
+    
+    // New fields
+    Labels      map[string]string `json:"labels,omitempty"`
+    Annotations map[string]string `json:"annotations,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+When a BuildRun executes for a Build that specifies `labels` or `annotations`, the BuildRun controller will append a new step to the end of the Tekton TaskRun that augments the image with the specified labels and annotations, and pushes to the same tag.
+
+This could be accomplished using a containerized CLI tool like [`crane`](https://github.com/google/go-containerregistry/blob/main/cmd/crane), or a purpose-built binary that Shipwright provides and packages in our releases.
```

Comment:
I personally would prefer more details, but I also see that the EP is marked "only" provisional. So, then it's probably fine.

Reply:
Added more detail.